### PR TITLE
Use `go.uber.org/atomic` everywhere

### LINKED ISSUE #5200: Decide on using or not uber/atomic vs golang standard atomic package
For more context about the discussion, see https://github.com/open-telemetry/opentelemetry-collector/pull/5181#discussion_r848174797

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:**

- Add depguard to enforce usage of `go.uber.org/atomic`
- Use `go.uber.org/atomic` everywhere (WIP, still need to fix some tests)

**Link to tracking Issue:** Fixes #5200

### CODE REVIEW

**Comment by mx-psi** [comment]
This is a WIP, I broke some tests which need fixing

File: .golangci.yml [hunk 1]
```diff
@@ -95,8 +95,16 @@ linters-settings:
       - kilometre
       - kilometres
 
+  depguard:
+    list-type: denylist
+    include-go-root: true
+    packages-with-error-message:
+      # See https://github.com/open-telemetry/opentelemetry-collector/issues/5200 for rationale
+      - sync/atomic: "Use go.uber.org/atomic instead of sync/atomic"
```
Comment (codeboten) [inline .golangci.yml:103]: nice! I've not used `depguard` before

**Review by bogdandrutu — APPROVED** [review]

**Review by codeboten — APPROVED** [review]
Change looks good, should the same be applied to the contrib repo as well?

**Comment by mx-psi** [comment]
> Change looks good, should the same be applied to the contrib repo as well?

I added an item to open-telemetry/opentelemetry-collector-contrib/issues/9109!

File: .golangci.yml (hunk 1 above)
Comment (mx-psi) [reply .golangci.yml:103]: Me neither, thanks to @hanjm for proposing it here https://github.com/open-telemetry/opentelemetry-collector/pull/5181#discussion_r848974947 :)

**Pull request merged**